api/enki.py

```diff
@@ -0,0 +1,633 @@
+            content = availability.content
+            count = 0
+            for bibliographic, circulation in BibliographicParser().process_all(
+                    content, "//enki:title"):
+                self.process_book(bibliographic, circulation)
+                count += 1
+                if count % self.batch_size == 0:
+                    self._db.commit()
+            x += step
+
+    def process_book(self, bibliographic, availability):
+
+        license_pool, new_license_pool = availability.license_pool(self._db)
+        edition, new_edition = bibliographic.edition(self._db)
+        license_pool.edition = edition
+        policy = ReplacementPolicy(
+            identifiers=False,
+            subjects=True,
+            contributions=True,
+            formats=True,
+        )
+        availability.apply(
+            pool=license_pool,
+            replace=policy,
+        )
+        if new_edition:
+            bibliographic.apply(edition, replace=policy)
+
+        if new_license_pool or new_edition:
+            # At this point we have done work equivalent to that done by
+            # the EnkiBibliographicCoverageProvider. Register that the
+            # work has been done so we don't have to do it again.
+            identifier = edition.primary_identifier
+            self.bibliographic_coverage_provider.handle_success(identifier)
+            self.bibliographic_coverage_provider.add_coverage_record_for(
+                identifier
+            )
+
+        return edition, license_pool
+
+#Copied from 3M. Eventually we might want to refactor
+class EnkiCollectionReaper(IdentifierSweepMonitor):
+    """Check for books that are in the local collection but have left the Enki collection."""
+    def __init__(self, _db, api=None, interval_seconds=3600*4):
+        super(EnkiCollectionReaper, self).__init__(_db, "Enki Collection Reaper", interval_seconds)
+        self._db = _db
+        if not api:
+            api = EnkiAPI.from_environment(_db)
+        self.api = api
+        self.data_source = DataSource.lookup(self._db, DataSource.ENKI)
+
+    def run(self):
+        self.api = EnkiAPI.from_environment(self._db)
+        self.data_source = DataSource.lookup(self._db, DataSource.ENKI)
+        super(EnkiCollectionReaper, self).run()
+
+    def identifier_query(self):
+        return self._db.query(Identifier).filter(
+            Identifier.type==Identifier.ENKI_ID)
+
+    def process_batch(self, identifiers):
+        enki_ids = set()
+        for identifier in identifiers:
+            enki_ids.add(identifier.identifier)
+
+        identifiers_not_mentioned_by_enki= set(identifiers)
+        now = datetime.datetime.utcnow()
+
+        for identifier in identifiers:
+            result = self.api.reaper_request(identifier.identifier)
+            if not result:
+                print "skipping this deleted book"
+                continue
+            print "keeping this existing book"
```

Comment:
These print statements should be removed IMO. `reaper_request` contains the informatino you would want to keep in the logs.